### PR TITLE
Update ReadMe And How to build a duck

### PR DESCRIPTION
**Pull request by TimoWielink**
Changed the way we use the ReadMe and How to build a Duck

With the changes made over the last couple of days, we now have support for PlatformIO. 
By adding #30 we now can reference installation instructions to  doc/how to build a duck and clean up the main Readme

Added Installation Section
- Moved Arduino Ide Installation to How to build a Duck
- Moved PlatformIO  Installation to How to build a Duck
- Moved IBM Watson IoT platform  Installation to How to build a Duck
- Fix Project OWL Website links

Updated How to build a duck 
-  Added installation for Submodule Libraries
- Added PlatformIO section
-Added IBM Watson IoT section
- Updated Duck Pictures

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,46 +19,16 @@ This is beneficial after events such as earthquakes or hurricanes where communic
...
-# API and Getting Started
-Check out the [ClusterDuck Protocol] website for more information and to learn about projects built upon this codebase. You can reach out directly on our [Slack] too! Check out our [How To Build A Duck User Manual] for in depth instructions for setting up your environment and materials. 
 
-## PlatformIO
-[PlatformIO](https://platformio.org/) is a ecosystem for embedded development. Grab your favorite IDE from [here](https://platformio.org/install/integration) and make sure to install the platformIO ide extension. If you are not sure which IDE to use, use [VSCode](https://docs.platformio.org/en/latest/integration/ide/vscode.html#installation).
+# Installation
+Check out the [ClusterDuck Protocol] website for more information and to learn about projects built upon this codebase. You can reach out directly on our [Slack] too for any questions! 
 
-### Installing ClusterDuck-Protocol globally
+To use the ClusterDuck Protocol follow the [How To Build a Duck](doc/howToBuildADuck.md) instalation guide.
```
Comment (krook) [inline README.md:26]: One letter typo on "installation"

File: README.md [hunk 2]
```diff
@@ -19,46 +19,16 @@ This is beneficial after events such as earthquakes or hurricanes where communic
 
 ![portal](https://github.com/knouse1344/owl/blob/master/app/assets/images/cluster_demo_vector.gif)
 
-# API and Getting Started
-Check out the [ClusterDuck Protocol] website for more information and to learn about projects built upon this codebase. You can reach out directly on our [Slack] too! Check out our [How To Build A Duck User Manual] for in depth instructions for setting up your environment and materials. 
 
-## PlatformIO
-[PlatformIO](https://platformio.org/) is a ecosystem for embedded development. Grab your favorite IDE from [here](https://platformio.org/install/integration) and make sure to install the platformIO ide extension. If you are not sure which IDE to use, use [VSCode](https://docs.platformio.org/en/latest/integration/ide/vscode.html#installation).
+# Installation
+Check out the [ClusterDuck Protocol] website for more information and to learn about projects built upon this codebase. You can reach out directly on our [Slack] too for any questions!
```
Comment (krook) [inline README.md:24]: I would say [Slack workspace]

File: README.md [hunk 3]
```diff
@@ -234,7 +183,7 @@ This project is licensed under the Apache 2 License - see the [LICENSE](LICENSE)
 v1.0.5
 
 
-[Project OWL]: <https://project-owl.com>
+[Project OWL]: <https://www.project-owl.com/>
 [ClusterDuck Protocol]: <http://clusterduckprotocol.org>
 [Slack]: <https://www.project-owl.com/slack>
```
Comment (krook) [inline README.md:188]: If we change line 24 to [Slack workspace] we would change it here too

File: doc/howToBuildADuck.md [hunk 4]
```diff
@@ -51,6 +55,8 @@ PapaDuck is the final Duck in the ClusterDuck and transmits network data to the
 
 You will need the [Arduino IDE](https://www.arduino.cc/en/main/software) to install the firmware on your Arduino devices.
 
+Or you can use your favourite Code editor with the [platformIO](https://platformio.org/) plugin.
```
Comment (krook) [inline doc/howToBuildADuck.md:58]: American English is "favorite" without the "u"

File: doc/howToBuildADuck.md [hunk 5]
```diff
@@ -67,7 +73,21 @@ There are many combinations of electronics, batteries, and enclosures that can b
 
 * **Enclosure:** Plastic Box (or 3D print, or rubber ducky, or anything you want really) if a case is necessary, [this](https://www.amazon.com/gp/product/B07DVS1HC4/ref=ppx_yo_dt_b_asin_title_o06_s00?ie=UTF8&psc=1) would work.
 
-## Arduino Software Preparation
+## PlatformIO
+[PlatformIO](https://platformio.org/) is a ecosystem for embedded development. Grab your favorite IDE from [here](https://platformio.org/install/integration) and make sure to install the platformIO ide extension. If you are not sure which IDE to use, use [VSCode](https://docs.platformio.org/en/latest/integration/ide/vscode.html#installation).
```
Comment (krook) [inline doc/howToBuildADuck.md:77]: "ide" should be "IDE". Also, we should be globally consistent about "PlatformIO" vs "platformIO".

File: README.md [hunk 6]
```diff
@@ -120,28 +90,7 @@ Now compile and upload to your device. If using a Heltec LoRa ESP32 board you sh
 
 If you don't see the captive portal screen, you can force it by accessing [neverssl.com](http://neverssl.com) which will force the captive portal to intercept the HTTP request.
```
Comment (krook) [inline README.md:91]: "which will help the captive portal intercept the HTTP request"

**Comment by Salzi** [comment]
Just three general comments:

- Currently you have at least 3 different sources for assets: this repo, the repo over from project owl and the project ow website. It would probably make sense to have all assets centralized here in case something is not reachable or has changed

- Personally, I don't think it's a good idea to do an API description by hand. There are several tools to do that. I made a suggestion in PR #31 . Doxygen is used. This has several advantages: The user can see the comments in the IDE while developing his ducks, for you as a developer the documentation is created where it takes place, namely directly at the API function. Doxygen also notices when a parameter changes and warns if it has not been documented yet etc.

- To keep your repo small you should think about Git LFS (for pictures, PDFs, zips, ...)

File: README.md (hunk 1 above)
Comment (TimoWielink) [reply README.md:26]: solved

File: README.md (hunk 2 above)
Comment (TimoWielink) [reply README.md:24]: solved

File: README.md (hunk 3 above)
Comment (TimoWielink) [reply README.md:188]: solved

File: doc/howToBuildADuck.md (hunk 4 above)
Comment (TimoWielink) [reply doc/howToBuildADuck.md:58]: solved

File: doc/howToBuildADuck.md (hunk 5 above)
Comment (TimoWielink) [reply doc/howToBuildADuck.md:77]: solved

File: README.md (hunk 6 above)
Comment (TimoWielink) [reply README.md:91]: solved

**Comment by TimoWielink** [comment]
@Salzi very good points

I will change the Files and ad LFS , for the second one I will pull in @nfeuer and take a close look at your PR

**Comment by Salzi** [comment]
I had the feeling that the team was not completely convinced of the auto generated solution. Because you have to upload the files to a server by hand. Therefore I played a bit with github action. Here is my solution (would recommend to integrate it here instead of having another PR).

Change the file `.github/workflows/documentation.yml` to

```yaml
name: Documentation

on: push

jobs:
  convert_via_pandoc:
    runs-on: ubuntu-18.04
    steps:
      - uses: actions/checkout@v2
      - run: mkdir output
      - uses: Ender-events/pandoc@master
        with: 
          args: "--resource-path=doc -f markdown-implicit_figures doc/howToBuildADuck.md --pdf-engine=xelatex  -V \"mainfont:DejaVu Sans\" -V \"geometry: left=2cm,right=2cm,top=3cm,bottom=3cm\" -o output/howToBuildADuck.pdf"
      - uses: docker://michaelperrin/pdfunite:0.48.0
        with:
          args: "doc/assets/pdfs/title.pdf output/howToBuildADuck.pdf output/final.pdf"
      - run: rm output/howToBuildADuck.pdf
      - run: cp output/final.pdf doc/HowToBuildADuck.pdf
      - uses: actions/upload-artifact@master
        with:
          name: output
          path: output
      - run: rm -rf output
      - uses: mikeal/publish-to-github-action@master
        if: github.ref == 'refs/heads/master'
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

The main difference is the last step:

```yaml
      - uses: mikeal/publish-to-github-action@master
        if: github.ref == 'refs/heads/master'
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
This will push the generated pdf back to the repository but only when running on the master branch. So basically you create a branch, edit the how to and push it to the repository. The github action will generate the pdf and stores it as artifact. You can download it and check it. If everything is ok merge it with master and the action will push the pdf to your master branch. You can see the running setup [here](https://github.com/Salzi/ClusterDuck-Protocol/blob/master/doc/HowToBuildADuck.pdf) (check _Automated Publisher_). So on your websites you can always link to the pdf on your master branch and it would be always up to date.

With this solution it's even more important to use git lfs. So at least at a `.gitattributes` file with:

```
*.pdf filter=lfs diff=lfs merge=lfs -text
```
to your repository. (Better would be to add all other binary formats too. Like: *.png, *.jpg, *.jpeg, *.psd ...)

Also consider using `git lfs migrate` to clean up your repository.

**Comment by TimoWielink** [comment]
Hi @Salzi seems like a great solution let me get on it now

**Comment by TimoWielink** [comment]
- [ ] Install and migrate to LFS

- [ ] Test and validate Pandoc pdf generator 

- [ ] Update image sources

**Pull request merged**